pkg/operator/ceph/cluster/osd/spec_test.go

```diff
@@ -450,6 +450,19 @@ func testPodDevices(t *testing.T, dataDir, deviceName string, allDevices bool) {
 		assert.True(t, shareProcessNamespace)
 	}
 
+	t.Run(("check osd ConfigureProbe"), func(t *testing.T) {
+		c.spec.HealthCheck.StartupProbe = make(map[cephv1.KeyType]*cephv1.ProbeSpec)
+		c.spec.HealthCheck.StartupProbe[cephv1.KeyOSD] = &cephv1.ProbeSpec{Disabled: false, Probe: &v1.Probe{InitialDelaySeconds: 1000}}
+		c.spec.HealthCheck.LivenessProbe = make(map[cephv1.KeyType]*cephv1.ProbeSpec)
+		c.spec.HealthCheck.LivenessProbe[cephv1.KeyOSD] = &cephv1.ProbeSpec{Disabled: false, Probe: &v1.Probe{InitialDelaySeconds: 900}}
```

Comment:
Hey, @BlaineEXE thanks for this amazing suggestion and guidance.

I tried to use the Deepcopy method, but I didn't see if I can directly copy the cluster, but instead I am getting suggestions for copying cluster spec, etc  `c.spec.DeepCopy()`, ig if it would be like this then there will be a lot more to do, to create a copy of a cluster.
I am not sure how we can update the autogenerated function in the Deepcopy file, AFAIU the mgr cluster is created internally with the rook, and not with help of crds generation, right?